audio_service/android/src/main/java/com/ryanheise/audioservice/AudioServicePlugin.java

```diff
@@ -49,197 +50,203 @@
 import android.net.Uri;
 
 /**
- * AudioservicePlugin
+ * A connection bridge between {@link AudioService} and dart,
+ * therefore creates and manages a {@link FlutterEngine} instance.
+ *
+ * With that mechanism AudioService can be start alone, creating the plugin, and then later
+ * FlutterActivity attached to it, as well as FlutterActivity may be launched first
+ * and create the AudioService.
  */
 public class AudioServicePlugin implements FlutterPlugin, ActivityAware {
-    private static String flutterEngineId = "audio_service_engine";
-    /** Must be called BEFORE any FlutterEngine is created. e.g. in Application class. */
-    public static void setFlutterEngineId(String id) {
-        flutterEngineId = id;
-    }
-    public static String getFlutterEngineId() {
-        return flutterEngineId;
-    }
-    public static synchronized FlutterEngine getFlutterEngine(Context context) {
-        FlutterEngine flutterEngine = FlutterEngineCache.getInstance().get(flutterEngineId);
-        if (flutterEngine == null) {
-            // XXX: The constructor triggers onAttachedToEngine so this variable doesn't help us.
-            // Maybe need a boolean flag to tell us we're currently loading the main flutter engine.
-            flutterEngine = new FlutterEngine(context.getApplicationContext());
-            flutterEngine.getDartExecutor().executeDartEntrypoint(DartExecutor.DartEntrypoint.createDefault());
-            FlutterEngineCache.getInstance().put(flutterEngineId, flutterEngine);
```

Comment:
It can be avoided (see Application class in the example), but I'm not really sure why is this a problem